server/src/main/java/io/crate/metadata/functions/SignatureBinder.java

```diff
@@ -345,7 +356,7 @@ private BoundVariables iterativeSolve(List<TypeConstraintSolver> constraints) {
     }
 
     private boolean allTypeVariablesBound(BoundVariables boundVariables) {
-        return boundVariables.getTypeVariableNames().equals(typeVariableConstraints.keySet());
```

Comment:
This change is due to the fact that the bindings may now include parametrized types (eg `numeric=>numeric(10)`) that are not included as type variables in the signature declaration.

Reply:
Could you please add this as a comment in the code, I think it could be useful for someone looking into this in the future.